src/triangle.rs

```diff
@@ -0,0 +1,344 @@
+            a: Point::new(a.0, a.1),
+            b: Point::new(b.0, b.1),
+            c: Point::new(c.0, c.1),
+        }
+    }
+
+    /// The centroid of the [`Triangle`]
+    #[inline]
+    pub fn centroid(&self) -> Point {
+        (1.0 / 3.0 * (self.a.to_vec2() + self.b.to_vec2() + self.c.to_vec2())).to_point()
+    }
+
+    /// The circumcenter of the [`Triangle`]
+    #[inline]
+    pub fn circumcenter(&self) -> Point {
+        let d = 2.0
+            * (self.a.x * (self.b.y - self.c.y)
+                + self.b.x * (self.c.y - self.a.y)
+                + self.c.x * (self.a.y - self.b.y));
+
+        let ux = ((self.a.x.powi(2) + self.a.y.powi(2)) * (self.b.y - self.c.y)
+            + (self.b.x.powi(2) + self.b.y.powi(2)) * (self.c.y - self.a.y)
+            + (self.c.x.powi(2) + self.c.y.powi(2)) * (self.a.y - self.b.y))
+            / d;
+
+        let uy = ((self.a.x.powi(2) + self.a.y.powi(2)) * (self.c.x - self.b.x)
+            + (self.b.x.powi(2) + self.b.y.powi(2)) * (self.a.x - self.c.x)
+            + (self.c.x.powi(2) + self.c.y.powi(2)) * (self.b.x - self.a.x))
+            / d;
+
+        Point::new(ux, uy)
+    }
+
+    /// The offset of each vertex from the centroid
+    #[inline]
+    pub fn offsets(&self) -> [Vec2; 3] {
+        let centroid = self.centroid().to_vec2();
+
+        [
+            (self.a.to_vec2() - centroid),
+            (self.b.to_vec2() - centroid),
+            (self.c.to_vec2() - centroid),
+        ]
+    }
+
+    /// The area of the [`Triangle`]
+    #[inline]
+    pub fn area(&self) -> f64 {
+        0.5 * (self.b - self.a).cross(self.c - self.a)
+    }
+
+    /// Whether this [`Triangle`] has no (zero) area
+    #[inline]
+    pub fn is_empty(&self) -> bool {
+        self.area() == 0.0
+    }
+
+    /// Inradius of [`Triangle`] (the greatest radius of a circle that is within the [`Triangle`])
+    /// with center [`Triangle::circumcenter`]
+    #[inline]
+    pub fn inradius(&self) -> f64 {
+        let ab = self.a.distance(self.b);
+        let bc = self.b.distance(self.c);
+        let ac = self.a.distance(self.c);
+
+        2.0 * self.area() / (ab + bc + ac)
+    }
+
+    /// Circumradius of [`Triangle`] (the smallest radius of a circle such that it intercepts each vertex)
+    /// with center [`Triangle::circumcenter`]
+    #[inline]
+    pub fn circumradius(&self) -> f64 {
+        let ab = self.a.distance(self.b);
+        let bc = self.b.distance(self.c);
+        let ac = self.a.distance(self.c);
+
+        (ab * bc * ac) / (4.0 * self.area())
+    }
+
+    /// Expand the triangle by a constant amount (`scalar`) in all directions
+    pub fn inflate(&self, scalar: f64) -> Self {
+        let centroid = self.centroid();
+
+        Self::new(
+            centroid + (0.0, scalar),
+            centroid + scalar * Vec2::from_angle(5.0 * FRAC_PI_4),
+            centroid + scalar * Vec2::from_angle(7.0 * FRAC_PI_4),
+        )
+    }
+
+    /// Is this [`Triangle`] finite?
```

Comment:
Please update this and the `is_nan` doc comments to match the newer style on the other types.

Reply:
just saw #371 should i add it here too?

Reply:
Yes please!